From 80e4e5ba7260f5ac23c0e74f625b04ebbe189f5b Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Thu, 2 May 2013 19:09:42 +0300
Subject: [PDK MR2 Temp] Handling of correct data profile

BZ: 105157

This patch is a temporary fix to use the MMS apn profile.
The actual fix would have to be done RILConstants.java, which
is not available in PDK.

Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>

Change-Id: I809f1e1df59ae5643d0f1632b912fb5a250e06ae
---
 .../telephony/dataconnection/DcTracker.java        |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index bbe10eb..38faa28 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -2107,6 +2107,14 @@ public final class DcTracker extends DcTrackerBase {
             return RILConstants.DATA_PROFILE_FOTA;
         } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_CBS)) {
             return RILConstants.DATA_PROFILE_CBS;
+        } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_MMS)) {
+            return (RILConstants.DATA_PROFILE_CBS + 1);
+        } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_DUN)) {
+            return RILConstants.DATA_PROFILE_TETHERED;
+        } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_SUPL)) {
+            return (RILConstants.DATA_PROFILE_CBS + 2);
+        } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_HIPRI)) {
+            return (RILConstants.DATA_PROFILE_CBS + 3);
         } else {
             return RILConstants.DATA_PROFILE_DEFAULT;
         }
-- 
1.7.4.1

